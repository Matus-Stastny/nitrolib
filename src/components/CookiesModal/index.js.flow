// @flow strict
import * as React from "react";

type Props = {|
  onClose: () => void,
|};

declare export default React.ComponentType<Props>;
