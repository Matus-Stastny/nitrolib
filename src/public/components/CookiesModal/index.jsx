// @flow strict
import * as React from "react";
import styled from "styled-components";

import Modal from "public/components/Modal/index";
import Text from "public/components/Text/index";
import { Consumer as BrandConsumer } from "public/services/brand/context";

const Container = styled.div``;

const Title = styled.h1``;

const Content = styled.p``;

const tKeys = {
  kayakwhite: __("content.cookies.kayakwhite"),
  kiwicom: __("content.cookies.kiwicom"),
  paragraphs: __("content.cookies.paragraphs"),
};

type Props = {|
  onClose: () => void,
|};

const CookiesModal = ({ onClose }: Props) => (
  <Modal onClose={onClose}>
    <Container>
      <Title>
        <Text t={__("content.pages.cookies.title")} />
      </Title>
      <Content>
        <BrandConsumer>
          {brand => <Text t={tKeys[brand.content.legal.cookies.t_key]} html />}
        </BrandConsumer>
      </Content>
    </Container>
  </Modal>
);

export default CookiesModal;
