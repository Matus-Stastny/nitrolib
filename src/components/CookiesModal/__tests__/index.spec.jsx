// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import CookiesModal from "..";

describe("#CookiesModal", () => {
  test("render", () => {
    const wrapper = shallow(<CookiesModal onClose={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("pass close", () => {
    const onClose = jest.fn();
    const wrapper = shallow(<CookiesModal onClose={onClose} />);

    expect(wrapper.find("Modal").prop("onClose")).toBe(onClose);
  });
});
