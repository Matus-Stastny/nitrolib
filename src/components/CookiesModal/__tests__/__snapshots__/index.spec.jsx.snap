// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#CookiesModal render 1`] = `
<Modal
  onClose={[MockFunction]}
>
  <CookiesModal__Container>
    <CookiesModal__Title>
      <Text
        html={false}
        t="content.pages.cookies.title"
        values={Object {}}
      />
    </CookiesModal__Title>
    <CookiesModal__Content>
      <[object Object] />
    </CookiesModal__Content>
  </CookiesModal__Container>
</Modal>
`;
